sdk/src/main/scala/com/horizen/account/block/AccountBlockHeader.scala

```diff
@@ -141,9 +165,9 @@ object AccountBlockHeaderSerializer extends SparkzSerializer[AccountBlockHeader]
     w.putInt(baseFee.length)
     w.putBytes(baseFee)
 
-    w.putLong(obj.gasUsed)
-
-    w.putLong(obj.gasLimit)
+    // longValueExact throws exception if overflows
+    w.putLong(obj.gasUsed.longValueExact())
+    w.putLong(obj.gasLimit.longValueExact())
```

Comment:
Do we want these fields to be serialized as before or as `baseFee` above?
In case for some reasons the value stored here will exceed `long` ranges, we may require an ugly workaround.
to be discussed.